georchestra-authnz/src/main/java/org/georchestra/geonetwork/security/integration/AccountsSynchroizationService.java

```diff
@@ -0,0 +1,90 @@
+//===	the Free Software Foundation; either version 2 of the License, or (at
+//===	your option) any later version.
+//===
+//===	This program is distributed in the hope that it will be useful, but
+//===	WITHOUT ANY WARRANTY; without even the implied warranty of
+//===	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+//===	General Public License for more details.
+//===
+//===	You should have received a copy of the GNU General Public License
+//===	along with this program; if not, write to the Free Software
+//===	Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+//===
+//===	Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+//===	Rome - Italy. email: geonetwork@osgeo.org
+//==============================================================================
+package org.georchestra.geonetwork.security.integration;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.georchestra.geonetwork.logging.Logging;
+import org.georchestra.security.api.OrganizationsApi;
+import org.georchestra.security.api.UsersApi;
+import org.georchestra.security.model.GeorchestraUser;
+import org.georchestra.security.model.Organization;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ApplicationContextEvent;
+import org.springframework.context.event.ContextStartedEvent;
+import org.springframework.context.event.ContextStoppedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AccountsSynchroizationService implements ApplicationListener<ApplicationContextEvent> {
+
+    private static final Logging log = Logging.getLogger("org.georchestra.geonetwork.security.integration");
+
+    private @Autowired UsersApi usersApi;
+    private @Autowired OrganizationsApi organizationsApi;
+    private @Autowired GeorchestraToGeonetworkUserReconcilingService reconcilingService;
+
+    private ScheduledExecutorService scheduledExecutor;
+
+    @Override
+    public void onApplicationEvent(ApplicationContextEvent event) {
+        if (event instanceof ContextStartedEvent) {
+            synchronize();
+            log.info("Users and groups synhchronized upon application startup.");
+            log.info("Setting up scheduled synchronization every 1 minute");
+            this.scheduledExecutor = Executors.newScheduledThreadPool(1);
+            this.scheduledExecutor.scheduleWithFixedDelay(this::synchronize, 1, 1, TimeUnit.MINUTES);
+        } else if (event instanceof ContextStoppedEvent) {
+            if (this.scheduledExecutor != null) {
+                this.scheduledExecutor.shutdownNow();
+            }
+        }
```

Comment:
Didn't bother to go with quarks, since a longer term goal would be to get changes in an event stream as soon as they happen in console, instead of having to poll.